Library/Formula/caf.rb

```diff
@@ -0,0 +1,52 @@
+class Caf < Formula
+  homepage "http://actor-framework.org"
+  head "https://github.com/actor-framework/actor-framework.git"
+
+  stable do
+    url "https://github.com/actor-framework/actor-framework/archive/0.12.0.tar.gz"
+    sha256 "294fb00508c166595d479e9af2fae28f2d370ed164cd1de079ae247acc916801"
+  end
+
+  devel do
+    url "https://github.com/actor-framework/actor-framework.git",
+      :branch => "develop"
+    version "0.13-devel"
+  end
+
+  depends_on "cmake" => :build
+
+  option "with-static", "build as static and shared library"
+  option "with-static-only", "build as static library only"
+  option "with-more-warnings", "build with most compiler warnings"
+  option "with-warnings-as-errors", "elevate compiler warnings to errors"
+  option "without-compiler-check", "disable compiler version check"
+  option "with-runtime-checks", "build with assertions at runtime"
+  option "with-log-level-error", "build with log level ERROR"
+  option "with-log-level-warning", "build with log level WARNING"
+  option "with-log-level-info", "build with log level INFO"
+  option "with-log-level-debug", "build with log level DEBUG"
+  option "with-log-level-trace", "build with log level TRACE"
+
+  def install
+    args = ["./configure", "--prefix=#{prefix}", "--no-examples"]
+    args << "--build-static" if build.with? "static"
+    args << "--build-static-only" if build.with? "static-only"
+    args << "--more-warnings" if build.with? "more-warnings"
+    args << "--warnings-as-errors" if build.with? "warnings-as-errors"
+    args << "--no-compiler-check" if build.without? "compiler-check"
+    args << "--with-runtime-checks" if build.with? "runtime-checks"
+    args << "--with-log-level=ERROR" if build.with? "log-level-error"
+    args << "--with-log-level=WARNING" if build.with? "log-level-warning"
+    args << "--with-log-level=INFO" if build.with? "log-level-info"
+    args << "--with-log-level=DEBUG" if build.with? "log-level-debug"
+    args << "--with-log-level=TRACE" if build.with? "log-level-trace"
+
+    # Clear state from previous builds to allow for changing build options.
+    system "make", "distclean" if File.exist?("Makefile")
+
+    system *args
+    system "make"
+    system "make", "test"
+    system "make", "install"
+  end
+end
```

Comment:
Since this is a library, I wasn't sure how to go after the unit tests. CAF ships with an extensive suite of unit tests (`make test`), would it make sense to run them in `test do`? From skimming other formulae, it looked like `test do` mostly uses files from the already installed project, not from the build tree. Hence I skipped the test block.